wled00/FX_fcn.cpp

```diff
@@ -1201,15 +1319,20 @@ void WS2812FX::service() {
         [[maybe_unused]] uint8_t tmpMode = seg.currentMode();  // this will return old mode while in transition
         delay = (*_mode[seg.mode])();         // run new/current mode
 #ifndef WLED_DISABLE_MODE_BLEND
-        if (modeBlending && seg.mode != tmpMode) {
+        if (modeBlending && seg.isInTransition()) {
```

Comment:
Effect blending only happens when actual effects (mode) change.
This change is incorrect.

Reply:
Yeah, since I'm hooking into your effect blending implementation, a change to this line is necessary (however, it doesn't have to be exactly like this). In a previous version, I wrote

```
if (modeBlending && seg.isInTransition() && (seg.mode != tmpMode || transitionStyle != TRANSITION_STYLE_FADE)
```

which did work, but I noticed that changes to effects would ignore the selected transition style. Now, it's debatable if this is a bug or a feature, so I'm happy to revert back to the above change. Or as mentioned above, implement transition styles independent of the effect blending implementation.

Reply:
You got that wrong. This line does not need to change.
The code within `if` needs to execute only if effect blending is actually happening. The actual change needs to happen in the blending part (in `Segment::setPixelColor()`) but that might be challenging without introduction of "layering" (see my other posts on that matter). It will also require a bit more resources (and hence ESP32 only).